test/common/upstream/health_check_fuzz.proto

```diff
@@ -0,0 +1,51 @@
+syntax = "proto3";
+
+package test.common.upstream;
+
+import "validate/validate.proto";
+import "test/fuzz/common.proto";
+import "envoy/config/core/v3/health_check.proto";
+import "google/protobuf/empty.proto";
+
+message HttpRespond { //TODO: Split this across headers, body, and trailers?
+  test.fuzz.Headers headers = 1;
+  //https://github.com/nodejs/http-parser/blob/master/http_parser.c#L852
+  string status = 2 [(validate.rules).string.pattern = "[0-9]+$*"];
+}
+
+message Respond {
+  HttpRespond http_respond = 1;
+}
+
+message TriggerIntervalTimer {
+}
+
+message TriggerTimeoutTimer {
+}
+
+message RaiseEvent {
+  oneof event_selector {
+    option (validate.required) = true;
+    google.protobuf.Empty connected = 1;
+    google.protobuf.Empty remote_close = 2;
+    google.protobuf.Empty local_close = 3;
+  }
+}
+
+message Action {
+  oneof action_selector {
+    Respond respond = 1;
+    google.protobuf.Empty trigger_interval_timer = 2;
+    //TODO: respondBody, respondTrailers
+    google.protobuf.Empty trigger_timeout_timer = 3;
+    RaiseEvent raise_event = 4;
+  }
+}
+
+message HealthCheckTestCase {
+  envoy.config.core.v3.HealthCheck health_check_config = 1
+      [(validate.rules).message.required = true];
+  repeated Action actions = 2;
```

Comment:
did you find any need to cap events (in the code with MaxEvents or a protobuf constraint, if protobuf constraint, probably aim high since if it's too low and you want to extend, then existing testcases will be invalidated).

Reply:
Not so far, but I've only run fuzzer for single digit minutes. Do you want me to add?

Reply:
that, or you can leave for a follow up and just know that it might come up as a timeouts and the fix is as easy as adding a maximum in the for-loop.

Reply:
Added comment.